### PR TITLE
Dockerfile

### PR DESCRIPTION
**Pull request by FernandoMiguel**
Add a Dockerfile (and .dockerignore)
I used this to build for linux and darwin 
and anyone else that doesn't have golang on their system

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+## build with:
+## docker buildx build --platform=local --output dist .
+##
+## binary for your arch will be inside dist folder
+##
+## to build for OSX, run previously:
+## docker buildx create --use --platform darwin/amd64
+##
+
+FROM --platform=$BUILDPLATFORM tonistiigi/xx:golang AS xgo
```
Comment (rs) [inline Dockerfile]: Why do you need this?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ENV CGO_ENABLED=0
+COPY --from=xgo / /
+
+ARG TARGETPLATFORM
+RUN go env
+
+WORKDIR /src
+COPY . /src/
+
+RUN go build -o /go/bin/nextdns
```
Comment (rs) [inline Dockerfile]: You may want to add `-ldflags="-s -w"` as we do for binary distrib.

File: Dockerfile (hunk 1 above)
Comment (FernandoMiguel) [reply Dockerfile]: https://github.com/tonistiigi/xx/tree/master/golang
it's used to do cross build... I need that to build the mac bin on mac

File: Dockerfile (hunk 2 above)
Comment (FernandoMiguel) [reply Dockerfile]: I'm not familiar with this

File: Dockerfile (hunk 1 above)
Comment (FernandoMiguel) [reply Dockerfile]: otherwise it builds a linux bin

Comment (rs) [reply Dockerfile]: Resolved by mistake.

Go can do cross compilation natively with GOOS and GOARCH. I would drop this dep and just rely on go tooling.

Comment (FernandoMiguel) [reply Dockerfile]: when building on a mac it uses a linux vm, which then the arch is wrong for the built artifacts
this pushes go to use the right arch

File: Dockerfile (hunk 2 above)
Comment (rs) [reply Dockerfile]: It strips the binary to make it smaller. Check this post for more info: https://blog.filippo.io/shrink-your-go-binaries-with-this-one-weird-trick/

File: Dockerfile (hunk 1 above)
Comment (rs) [reply Dockerfile]: Ok got it.

File: Dockerfile (hunk 2 above)
Comment (FernandoMiguel) [reply Dockerfile]: added

**Comment by rs** [comment]
Thx

**Pull request merged**